include/par-res-kern_shmem.h

```diff
@@ -33,3 +33,129 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <shmem.h>
 
 extern void bail_out(int);
+
+/* Handle functions first and preprocessor symbols second,  *
+ * i.e. separately, to keep the code mildly readable and    *
+ * avoid duplicating any of the function signatures, which  *
+ * could lead to errors.                                    */
+
+static void prk_shmem_init(void) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
```

Comment:
I would prefer to see one instance of this logic at the top of the header file (or eventually in config.h if you move to autoconf):

```
#if defined(SHMEM_MAJOR_VERSION) && defined(SHMEM_MINOR_VERSION)
#  if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
#  define HAVE_OPENSHMEM_1_2
#  endif
#endif
```

And then the switches in each function would be:

```#ifdef HAVE_OPENSHMEM_1_2```